### ISSUE TITLE
desktop behaves differently than extensions (which work right)

### ISSUE THREAD

**Issue by ousia** [ruffle-rs/ruffle#6569]
### Describe the bug

https://free-culture.tk/lfc-presentation.swf contains the following functions to go to next and previous slide (which are triggered by **Left** and **Right** keys):

``` js
function previous_slide() {
  if (mode < 2) {
    if (mode == 0) {
      mode = 1;
    }
    snd.stop();
    if (0 >= position) {
      sound_position = 0;
    } else {
      sound_position = times[position - 1];
    }
    snd.start(sound_position / 1000.0, 0);
  }
}

function next_slide() {
  if (mode < 2) {
    if (mode == 0) {
      mode = 1;
    }
    snd.stop();
    sound_position = times[position + 1];
    if (sound_length < sound_position) {
      sound_position = sound_length - 500;
    }
    snd.start(sound_position / 1000.0, 0);
  }
}
```

Using both selfhosted and Firefox extension (nightly 2022-03-29) slides go one by one (either forwards or backwards).

Using the desktop program (both Linux binary and Windows with WineHQ),

1. `next_slide()` doesn’t advance to +1 slide (but to a slide with bigger page number).
2. `previous_slide()` doesn’t go to -1 slide (but always advances to a slide with a bigger page number).

I wonder why the desktop program is misbehaving, while the extension is doing fine.

Many thanks for your help.

### Expected behavior

`previous_slide()` and `next_slide()` should work when pressing **Left** and **Right** for the desktop program in the same way as in the extension.

### Affected platform

Desktop app

### Operating system

Fedora 35

### Browser

Firefox 98

**Comment by ousia** [comment]
Many thanks for your fix. @Herschel.
